src/libexpr/flake/call-flake.nix

```
@@ -43,7 +43,7 @@ let
 
           outputs = flake.outputs (inputs // { self = result; });
 
-          result = outputs // sourceInfo // { inherit inputs; inherit outputs; inherit sourceInfo; };
+          result = outputs // sourceInfo // { inherit inputs; inherit outputs; inherit sourceInfo; type = "flake"; };
```

```suggestion
          result = outputs // sourceInfo // { inherit inputs; inherit outputs; inherit sourceInfo; _type = "flake"; };
```

In Nixpkgs/lib, `isType`, `setType`, and the module system `mk*` constructors all use `._type`.
We also have `pkgs._type = "pkgs"` in Nixpkgs.
`.type` is only used for derivations, afaict.

---

Yes, but derivations are the only one of these that are generated by Nix itself. I think the reason the others use `_type` is precisely that `type` is reserved for Nix.

---

Maybe `_type` was chosen to avoid interfering with domain-specific `type` attributes, such as `(mkOption foo).type`.
Nix also uses `type` in a domain-specific way (as opposed to poor man's nominal typing) in the `fetchTree` arguments.
To illustrate this idea, I think it would make sense for `fetchTree` to return `{ _type = "tree"; type = "git"; ... }`.

---

I defer to @edolstra on the attr name here.

---

I guess `_type` is better because it has a smaller chance of clashing with user identifiers. Probably `derivation` should have used that as well.